config/initializers/geocoder_mapbox.rb

```diff
@@ -0,0 +1,24 @@
+# Mapbox relevance is defined from 0 to 1, where 0 is the least relevant and 1 is the most relevant
+# @see https://docs.mapbox.com/api/search/geocoding/#geocoding-response-object
+RELEVANCE_LIMIT = 1
+Geocoder::Lookup.get(:mapbox)
+
+Geocoder::Lookup::Mapbox.class_eval do
+  # @override: the original method do not implement the filter_relevant_features line
+  def results(query)
+    return [] unless data = fetch_data(query)
+    if data['features']
+      filtered_results = filter_relevant_features(data['features'])
+      sort_relevant_feature(filtered_results)
+    elsif data['message'] =~ /Invalid\sToken/
+      raise_error(Geocoder::InvalidApiKey, data['message'])
+    else
+      []
+    end
+  end
```

Comment:
If we do the filtering here, will it have some consequences when adding locations manually? E.g. that it will not match good enough? Or is that not a big concern?

Reply:
No, the manual addition / edit does not use this geocoder (it uses the one from mapbox-gl.js)

Reply:
Ah, true!